contracts/mainnet/connectors/compound/v3/helpers.sol

```diff
@@ -0,0 +1,210 @@
+				params.to != address(0),
+			"invalid market/token/to address"
+		);
+		bool isEth = params.token == ethAddr;
+		address token_ = isEth ? wethAddr : params.token;
+
+		TokenInterface tokenContract = TokenInterface(token_);
+
+		params.from = params.from == address(0) ? address(this) : params.from;
+		uint256 initialBal = CometInterface(params.market).borrowBalanceOf(
+			params.from
+		);
+
+		require(
+			TokenInterface(params.market).balanceOf(params.from) == 0,
+			"borrow-disabled-when-supplied-base"
+		);
+
+		_withdrawHelper(params.market, token_, params.from, params.to, amt_);
+
+		uint256 finalBal = CometInterface(params.market).borrowBalanceOf(
+			params.from
+		);
+		amt_ = sub(finalBal, initialBal);
+
+		if (params.to == address(this))
+			convertWethToEth(isEth, tokenContract, amt_);
+
+		setUint(params.setId, amt_);
+
+		amt = amt_;
+		setId = params.setId;
+	}
+
+	function _withdraw(BorrowWithdrawParams memory params)
+		internal
+		returns (uint256 amt, uint256 setId)
+	{
+		uint256 amt_ = getUint(params.getId, params.amt);
+
+		require(
+			params.market != address(0) &&
+				params.token != address(0) &&
+				params.to != address(0),
+			"invalid market/token/to address"
+		);
+
+		bool isEth = params.token == ethAddr;
+		address token_ = isEth ? wethAddr : params.token;
+
+		TokenInterface tokenContract = TokenInterface(token_);
+		params.from = params.from == address(0) ? address(this) : params.from;
+
+		uint256 initialBal = getAccountSupplyBalanceOfAsset(
+			params.from,
+			params.market,
+			token_
+		);
+		amt_ = amt_ == uint256(-1) ? initialBal : amt_;
+
+		if (token_ == getBaseToken(params.market)) {
+			uint256 balance = TokenInterface(params.market).balanceOf(
+				params.from
+			);
+			if (balance > 0) {
+				require(amt_ <= balance, "withdraw-amt-greater-than-supplies");
+			}
+		}
+
+		_withdrawHelper(params.market, token_, params.from, params.to, amt_);
+
+		uint256 finalBal = getAccountSupplyBalanceOfAsset(
+			params.from,
+			params.market,
+			token_
+		);
+		amt_ = sub(initialBal, finalBal);
+
+		if (params.to == address(this))
+			convertWethToEth(isEth, tokenContract, amt_);
+
+		setUint(params.setId, amt_);
+
+		amt = amt_;
+		setId = params.setId;
+	}
+
+	function _transfer(
+		address market,
+		address token,
+		address from,
+		address to,
+		uint256 amt
+	) internal {
+		if (from == address(0)) {
+			CometInterface(market).transferAsset(to, token, amt);
+		} else {
+			CometInterface(market).transferAssetFrom(from, to, token, amt);
+		}
+	}
+
+	function getAccountSupplyBalanceOfAsset(
+		address account,
+		address market,
+		address asset
+	) internal returns (uint256 balance) {
+		if (asset == getBaseToken(market)) {
+			//balance in base
+			balance = TokenInterface(market).balanceOf(account);
+		} else {
+			//balance in asset denomination
+			balance = uint256(
+				CometInterface(market).userCollateral(account, asset).balance
+			);
+		}
+	}
+
+	function setAmt(
```

Comment:
```suggestion
	function _setAmt(
```

Reply:
@Richa-iitr Also, we need to rename the function. setAmt has different meaning.

Reply:
Maybe `calculateFromAmount`

Reply:
done.